src/bulletproof/dealer.rs

```diff
@@ -141,13 +138,13 @@ impl<'a> DealerAwaitingPolyCommitments<'a> {
         let T_1: EdwardsPoint = poly_commitments.iter().map(|pc| pc.T_1_j).sum();
         let T_2: EdwardsPoint = poly_commitments.iter().map(|pc| pc.T_2_j).sum();
 
-        let mut keccak = Keccak::v256();
-        keccak.update(self.bit_challenge.z.as_bytes());
-        keccak.update(self.bit_challenge.z.as_bytes());
```

Comment:
Was doing it twice a bug? :O

Reply:
Nope, we're still doing it twice. It may confusing because `keccak_hash` works differently.